### PR TITLE
Add mcxn236 vref adc support

### PR DESCRIPTION
**Pull request by NeilChen93**
Add mcxn236 vref and adc support

### CODE REVIEW

File: boards/nxp/frdm_mcxn236/doc/index.rst [hunk 1]
```diff
@@ -68,6 +68,14 @@ The FRDM-MCXN236 board configuration supports the following hardware features:
 +-----------+------------+-------------------------------------+
 | WATCHDOG  | on-chip    | watchdog                            |
 +-----------+------------+-------------------------------------+
+| PWM       | on-chip    | pwm                                 |
++-----------+------------+-------------------------------------+
+| CTIMER    | on-chip    | counter                             |
```
Comment (kartben) [inline boards/nxp/frdm_mcxn236/doc/index.rst:73]: Put in a separate commit?

Comment (NeilChen93) [reply boards/nxp/frdm_mcxn236/doc/index.rst]: Thanks for reminder, removed PWM and CTIMER.

**Review by danieldegrasse — APPROVED** [review]

**Review by decsny — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Review by anangl — APPROVED** [review]
